### PR TITLE
Remove the call graph processor

### PR DESCRIPTION
**Pull request by JCapucho**
Uses the `GlobalUse` in glsl-out to check for function availability

### CODE REVIEW

File: src/back/glsl/mod.rs [hunk 1]
```diff
@@ -126,14 +123,16 @@ impl fmt::Display for Version {
...
     pub version: Version,
-    /// The name and stage of the entry point
+    /// The stage of the entry point
+    pub shader_stage: ShaderStage,
+    /// The name of the entry point
     ///
     /// If no entry point that matches is found a error will be thrown while creating a new instance
     /// of [`Writer`](struct.Writer.html)
-    pub entry_point: (ShaderStage, String),
+    pub entry_point: &'a str,
```
Comment (kvark) [inline src/back/glsl/mod.rs:135]: I'm not sure this is needed to borrow. This makes it impossible to deserialize `Options`, and we'd ideally want to use it directly in those `.param.ron` files.

File: src/proc/analyzer.rs [hunk 2]
```diff
@@ -91,6 +91,9 @@ impl FunctionInfo {
     pub fn expression_count(&self) -> usize {
         self.expressions.len()
     }
+    pub fn global_uses(&self) -> &[GlobalUse] {
```
Comment (kvark) [inline src/proc/analyzer.rs:94]: Can we perhaps expose this as a more high-level methiod? Something like
```rust
impl FunctionInfo {
  pub fn dominates_global_use(&self, other: &Self) -> bool {...}
}
```

**Review by kvark — COMMENTED** [review]
Looks good overall, just a few notes to address!

File: src/back/glsl/mod.rs (hunk 1 above)
Comment (JCapucho) [reply src/back/glsl/mod.rs:135]: It's possible to deserialize a struct with borrows

Comment (kvark) [reply src/back/glsl/mod.rs:135]: My concern here, and it's a minor-ish concern, is that you have to carry that `'options` lifetime through the code. It's polluting the definitions, and it doesn't get us anything. So the old choice with `String` was both simpler and more in-line with other backends.

**Review by kvark — APPROVED** [review]
Thank you!

**Pull request merged**